jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpFields.java

```diff
@@ -989,6 +990,33 @@ default Mutable add(HttpFields fields)
             return this;
         }
 
+        /**
+         * <p>Adds a field associated with a list of values.</p>
+         *
+         * @param name the name of the field
+         * @param list the List value of the field. If null the field is cleared.
+         * @return this builder
+         */
+        default Mutable add(String name, List<String> list)
+        {
+            Objects.requireNonNull(name);
+            if (list == null)
+                throw new IllegalArgumentException("null list");
+            if (list.isEmpty())
+                return this;
+            if (list.size() == 1)
+            {
+                String v = list.get(0);
+                if (v == null)
+                    return this;
+                return add(name, v);
+            }
+            HttpField field = new HttpField.MultiHttpField(name, list);
+            if (field.getValueList() == null)
```

Comment:
This line is strange, as we know the list has more than 1 element at this point.
How can be given to the field, and the field return `null`?
At least it deserves a comment, e.g. "In case all values are blanks".

Reply:
removed (see below)